Error in apollo-server-plugin-response-cache

<!--

Thanks for filing an issue on Apollo Server!

Before reporting a bug, please first check for existing or closed issues
and read the instructions for filing a bug report:

https://github.com/apollographql/apollo-server/blob/main/CONTRIBUTING.md#reporting-bugs

### This bug report should include:

- [ ] A short, but descriptive title. The title doesn't need "Apollo" in it.
- [ ] The package name and version of Apollo showing the problem.
- [ ] If applicable, the last version of Apollo where the problem did _not_ occur.
- [ ] The expected behavior.
- [ ] The actual behavior.
- [ ] A **simple**, runnable reproduction!
      Please make a GitHub repository that anyone can clone and run and see the
      problem. Other great ways to demonstrate a reproduction are using our
      CodeSandbox template (https://codesandbox.io/s/apollo-server), or re-mixing
      our Glitch template (https://glitch.com/~apollo-launchpad).

  Please make sure that you include the following information to ensure that your issue is actionable.

  If you don't follow the template, your issue may end up being closed without anyone looking at it carefully, because it is not actionable for us without the information in this template.

-->
Hello everyone,  I got this error
```
Error: Cannot find module 'apollo-server-types'
[1] Require stack:
[1] - /project/node_modules/apollo-server-plugin-response-cache/dist/ApolloServerPluginResponseCache.js
[1] - /project/node_modules/apollo-server-plugin-response-cache/dist/index.js
[1] - /project/packages/package1/src/server/start.ts
[1] - /project/packages/package1/src/server/index.ts
[1]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
[1]     at Function.Module._resolveFilename (/project/node_modules/tsconfig-paths/lib/register.js:75:40)
[1]     at Function.Module._load (internal/modules/cjs/loader.js:746:27)
[1]     at Module.require (internal/modules/cjs/loader.js:974:19)
[1]     at require (internal/modules/cjs/helpers.js:92:18)
[1]     at Object.<anonymous> (/project/node_modules/apollo-server-plugin-response-cache/src/ApolloServerPluginResponseCache.ts:8:1)
[1]     at Module._compile (internal/modules/cjs/loader.js:1085:14)
[1]     at Module._compile (/project/node_modules/source-map-support/source-map-support.js:547:25)
[1]     at Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
[1]     at Object.nodeDevHook [as .js] (/project/node_modules/ts-node-dev/lib/hook.js:63:13)
[1] [ERROR] 21:04:23 Error: Cannot find module 'apollo-server-types'
[1] Require stack:
[1] - /project/node_modules/apollo-server-plugin-response-cache/dist/ApolloServerPluginResponseCache.js
[1] - /project/node_modules/apollo-server-plugin-response-cache/dist/index.js
[1] - /project/packages/package1/src/server/start.ts
[1] - /project/packages/package1/src/server/index.ts

```

So, I look up the source code of `ApolloServerPluginResponseCache` in 'apollo-server-plugin-response-cache'.
From what I see, now package.json has [apollo-server-types](https://github.com/apollographql/apollo-server/blob/291c17e255122d4733b23177500188d68fac55ce/packages/apollo-server-plugin-response-cache/package.json#L26) as devdependencies. However, [ApolloServerPluginResponse](https://github.com/apollographql/apollo-server/blob/291c17e255122d4733b23177500188d68fac55ce/packages/apollo-server-plugin-response-cache/src/ApolloServerPluginResponseCache.ts#L298) js uses an enum called `CacheScope`, so an error seems to occur. Maybe this should be in dependencies, not devdependencies?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,8 @@ The version headers in this history reflect the versions of Apollo Server itself
 ## vNEXT
 
 - `apollo-server-core`: Fix build error when building with `@rollup/plugin-commonjs`. [PR #5797](https://github.com/apollographql/apollo-server/pull/5797)
+- `apollo-server-plugin-response-cache`: Add missing dependency on `apollo-server-types` (broken since v3.0.0). [Issue #5804](https://github.com/apollographql/apollo-server/issues/5804) [PR #5816](https://github.com/apollographql/apollo-server/pull/5816)
+
 ## v3.4.0
 
 - `apollo-server-core`: You can now specify your own `DocumentStore` (a `KeyValueStore<DocumentNode>`) for Apollo Server's cache of parsed and validated GraphQL operation abstract syntax trees via the new `documentStore` constructor option. **This replaces the `experimental_approximateDocumentStoreMiB` option.** You can replace `new ApolloServer({experimental_approximateDocumentStoreMiB: approximateDocumentStoreMiB, ...moreOptions})` with:

--- a/packages/apollo-server-plugin-response-cache/package.json
+++ b/packages/apollo-server-plugin-response-cache/package.json
@@ -21,9 +21,7 @@
   },
   "dependencies": {
     "apollo-server-caching": "file:../apollo-server-caching",
-    "apollo-server-plugin-base": "file:../apollo-server-plugin-base"
-  },
-  "devDependencies": {
+    "apollo-server-plugin-base": "file:../apollo-server-plugin-base",
     "apollo-server-types": "file:../apollo-server-types"
   },
   "peerDependencies": {

--- a/packages/apollo-server-plugin-response-cache/src/ApolloServerPluginResponseCache.ts
+++ b/packages/apollo-server-plugin-response-cache/src/ApolloServerPluginResponseCache.ts
@@ -5,9 +5,10 @@ import type {
 import type {
   GraphQLRequestContext,
   GraphQLResponse,
+  CacheHint,
+  ValueOrPromise,
 } from 'apollo-server-types';
 import { KeyValueCache, PrefixingKeyValueCache } from 'apollo-server-caching';
-import type { CacheHint, ValueOrPromise } from 'apollo-server-types';
 import { CacheScope } from 'apollo-server-types';
 
 // XXX This should use createSHA from apollo-server-core in order to work on

